common/chains/client/node.go

```diff
@@ -0,0 +1,318 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+const queryTimeout = 10 * time.Second
+
+var errInvalidChainID = errors.New("invalid chain id")
+
+var (
+	promPoolRPCNodeVerifies = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_verifies",
+		Help: "The total number of chain ID verifications for the given RPC node",
+	}, []string{"chainID", "nodeName"})
```

Comment:
Does `chainID` now include information about the chain family? Otherwise this doesn't really apply any more

Reply:
The existing [chainType](https://github.com/smartcontractkit/chainlink/blob/27a93bd5a56b9913492ea02a31cfa2863fb08516/core/config/chaintype.go#L12-L18) type doesn't seem to fully encapsulate the possibilities for chainFamily. Do we need to create another type `chainFamily` to enable this label?

Reply:
Okay, I think for now I'll go with a string for `chainFamily` to enable this

Reply:
Just pushed changes implementing this